Issue #11042: NavigableMapConfig.getProperty does not resolve environmental variables

ddelponte opened the pull request:
Please see issue #11042 for more details

ddelponte committed:
Issue #11042: NavigableMapConfig.getProperty does not resolve environmental variables

ddelponte committed:
Issue #11042: NavigableMapConfig.getProperty does not resolve environmental variables
• removed unused import

ddelponte committed:
Issue #11042: NavigableMapConfig.getProperty does not resolve environmental variables
• fixed a bug uncovered by a failing test in `DataSourcesGrailsPluginTests`.  Great tests!

graemerocher commented on grails-core/src/main/groovy/org/grails/config/DotNotatedKeyParser.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.Map;
+import java.util.StringTokenizer;
+
+public class DotNotatedKeyParser {
```
Please make the class package private. No need to expose this API publicly. Also is a new class needed? A static method of NavigableMapConfig would probably be enough.

ddelponte replied:
Addressed in commit f2c6ba03a3c9c3a4a2a292bf3aa59a854201bb33

ddelponte committed:
Issue #11042: NavigableMapConfig.getProperty does not resolve environmental variables
• addressed the following PR comment:
> Please make the class package private. No need to expose this API publicly. Also is a new class needed? A static method of NavigableMapConfig would probably be enough.
